runtime/doc/pack.txt

```diff
@@ -355,16 +356,27 @@ del({names})                                                  *vim.pack.del()*
                  be managed by |vim.pack|, not necessarily already added to
                  current session.
 
-get()                                                         *vim.pack.get()*
-    Get data about all plugins managed by |vim.pack|
+get({names}, {opts})                                          *vim.pack.get()*
+    Get data about plugins managed by |vim.pack|
+
+    Parameters: ~
+      • {names}  (`string[]?`) List of plugin names. Default: all plugins
+                 managed by |vim.pack|.
+      • {opts}   (`table?`) A table with the following fields:
+                 • {details} (`boolean`) Whether to include plugin details.
+                   Default `false`.
 
     Return: ~
         (`table[]`) A list of objects with the following fields:
-        • {spec} (`vim.pack.SpecResolved`) A |vim.pack.Spec| with defaults
-          made explicit.
+        • {spec} (`vim.pack.SpecResolved`) A |vim.pack.Spec| with resolved
+          `name`.
         • {path} (`string`) Plugin's path on disk.
         • {active} (`boolean`) Whether plugin was added via |vim.pack.add()|
           to current session.
+        • {details}? (`{ branches: string[], state: string, tags: string[] }`)
+          • `branches` lists available Git branches (first is default).
+          • `state` is a current Git state.
```

Comment:
example? is it a status string, or a dict? (showing an example or type in the this doc will answer this question for users)

Reply:
> s it a status string, or a dict? (showing an example or type in the this doc will answer this question for users)

It is shown higher that it is a string.



> if `state` is the commit, and never something like HEAD or a non-sha name, probably should just call it `rev` or `commit`. Git calls that a commit-id, but that's probably too pedantic. I lean towards `rev` because that's a bit more generalized but not too vague.

Currently it is indeed a (short) commit hash. Naming it with not VCS related name is a bit more forward compatible. Like after resolving #34765 a plugin can be local and can have its own notion of "state".

If taking only Git-related stuff, then It can be `rev`, I guess.

Reply:
> Like after resolving #34765 a plugin can be local and can have its own notion of "state".

`rev` might work for that? What would be examples of local "state" values?

Reply:
> `rev` might work for that?

Saying that "local plugin's revision is xxx" just doesn't feel right. But maybe I am reading too much into this.

> What would be examples of local "state" values?

Don't know, but at least leaving the door open shouldn't be a bad thing. Plus somewhat related: documentation at start of `*vim.pack*` and confirmation buffer already use "State" / "State before"  / "State after". Which was chosen specifically to be more future compatible. It can also be changed, of course, but these need at least to be consistent.